### PR TITLE
Developer tools

### PR DESCRIPTION
**Pull request by ntorresd**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This PR updates the package template files:
- .github/workflows/pkgdown.yaml
- tests/spelling.R
- tests/testthat/helper-state.R
- tests/testthat/setup-options.R
- tools/check.env

In accordance to the discussion in [November 2023 spring cleaning #101](https://github.com/epiverse-trace/packagetemplate/issues/101)

### CODE REVIEW

**Review by Bisaloo — APPROVED** [review]

**Review by Bisaloo — CHANGES_REQUESTED** [review]
Could you also update the other GitHub Actions workflows please?

**Comment by jpavlich** [comment]
> > Could you also update the other GitHub Actions workflows please?
> 
> Done. Please let me know whether we should keep line 47 in `test-coverage.yaml`. @jpavlich , do you remember why we added this to the workflow?

That was to be able to use development versions of rstan. Since we are sticking only to CRAN versions, I guess we don't need it anymore

File: tools/check.env [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Do not report if package size is larger than 5 megabytes
+_R_CHECK_PKG_SIZES_=false
+
+# Do not check Rd cross references
+_R_CHECK_RD_XREFS_=false
+
+# Do not report if package requires GNU make
+_R_CHECK_CRAN_INCOMING_NOTE_GNU_MAKE_=false
```
Comment (Bisaloo) [inline tools/check.env:8]:
```suggestion
_R_CHECK_CRAN_INCOMING_NOTE_GNU_MAKE_=true
```

My understanding of the code here seems to suggest it should be `true`. That's somewhat counter intuitive but can we give it a try please?

https://github.com/wch/r-source/blob/b12ffba7584825d6b11bba8b7dbad084a74c1c20/src/library/tools/R/check.R#L2960C2-L2960C2

**Comment by Bisaloo** [comment]
Clean diff is here: https://github.com/epiverse-trace/serofoi/compare/developer-tools2?expand=1. Can I force push it here?

**Comment by ntorresd** [comment]
Closes #86 and #101

**Pull request merged**